src/net.h

```diff
@@ -22,6 +21,25 @@
 #include "addrman.h"
 
 #include "gridcoin.h"
+
+
+// BOOST_FOREACH placeholders
+// Code taken from https://stackoverflow.com/questions/8542591
+// Should be replaced by range-based for loops !
+namespace std {
+    template<class T> T begin(std::pair<T, T> p){return p.first;}
+    template<class T> T end(std::pair<T, T> p){return p.second;}
+}
+template<class Iterator> std::reverse_iterator<Iterator> make_reverse_iterator(Iterator it)
+{return std::reverse_iterator<Iterator>(it);}
+template<class Range> std::pair<std::reverse_iterator<decltype(begin(std::declval<Range>()))>,
+                      std::reverse_iterator<decltype(begin(std::declval<Range>()))>> make_reverse_range(Range&& r)
+{return std::make_pair(make_reverse_iterator(begin(r)), make_reverse_iterator(end(r)));}
```

Comment:
this and above Looks like some black magic

Reply:
It does. Hopefully we'll get rid of it as soon as all `BOOST_FOREACH` will be removed from code